src/shared/logger/activation.ts

```diff
@@ -198,3 +202,62 @@ async function registerLoggerCommands(context: vscode.ExtensionContext): Promise
         )
     )
 }
+
+/**
+ * Deletes the older logs when there are too many or are using too much space.
+ */
+async function cleanLogFiles(): Promise<void> {
+    let files = await fs.readdir(logPath)
+
+    if (files.length > 100) {
+        deleteOldLogFiles(files, 10)
+        files = await fs.readdir(logPath)
+    }
+
+    let dirSize = 0
+    const oversizedFiles = []
+    for (const log of files) {
+        try {
+            const logSize = (await fs.stat(path.join(logPath, log))).size
+            dirSize += logSize
+            if (logSize > LOG_MAX_BYTES) {
+                oversizedFiles.push(log)
+            }
+        } catch (e) {
+            getLogger().error('Failed to access file details for %0', path.join(logPath, log), e)
+        }
+    }
+    // remove any single files over 100MB
+    if (oversizedFiles.length) {
+        for (const file of oversizedFiles) {
+            await fs.unlink(path.join(logPath, file))
```

Comment:
need try-catch around this. But instead, I think it's better to use deleteOldLogFiles(), something like this:
```suggestion
            deleteOldLogFiles(oversizedFiles, 0)
```